src/init.cpp

```diff
@@ -687,34 +687,35 @@ void ThreadImport(std::vector<boost::filesystem::path> vImportFiles)
     if (fReindex || fReindexFast)
     {
         CImportingNow imp;
-        int nFile = 0;
-        if (fReindexFast) uiInterface.InitMessage(_("Reindexing block headers from files..."));
+        CDiskBlockPos pos(0, 0);
         while (fReindexFast)
         {
-            CDiskBlockPos pos(nFile, 0);
             if (!boost::filesystem::exists(GetBlockPosFilename(pos, "blk")))
                 break; // No block files left to reindex
             FILE *file = OpenBlockFile(pos, true);
             if (!file) break; // This error is logged in OpenBlockFile
-            LogPrintf("Reindexing block file blk%05u.dat, headers-only...\n", (unsigned int)nFile);
+            uiInterface.InitMessage(tfm::format(_("Reindexing block headers from files...") + "(blk%05i.dat)", pos.nFile));
+            LogPrintf("Reindexing block file blk%05i.dat, headers-only...\n", pos.nFile);
             LoadBlocksFromExternalFile(file, &pos, /*loadHeadersOnly*/true);
-            nFile++;
+            pos.nFile++;
+            pos.nPos = 0;
         }
         if (fReindexFast)
             LogPrintf("Headers-only reindexing finished. Going on with blocks\n");
 
-        nFile = 0;
-        uiInterface.InitMessage(_("Reindexing block from files..."));
+        pos.nFile = 0;
+        pos.nPos = 0;
```

Comment:
Probably these two lines should stay in the `if (freindexFast)` branched code: in fact if that was false we never incremented pos and therefore no need to reset it to 0.0

Reply:
Modified accordingly.